tests/jballerina-unit-test/src/test/resources/test-src/workers/wait-actions-negative.bal

```diff
@@ -90,6 +90,50 @@ function waitForAllTest() {
     restRec2 result16 = wait {id: f1, name: f2, age: f4};
 }
 
+type WaitResult record {|
+    boolean|error producer;
+    boolean consumer;
+|};
+
+function waitForAllTest2() {
+    worker Producer1 returns boolean|error {
+        return true;
+    }
+
+    worker Producer2 returns boolean|error {
+        return true;
+    }
+
+    worker Consumer1 returns boolean {
+        return false;
+    }
+
+    worker Consumer2 returns boolean {
+        return false;
+    }
+
+    WaitResult res = wait {producer: Producer1|Producer2, consumer: Consumer1};
```

Comment:
If we introduce `&` here(Producer1&Producer2), it gives compile time error as shown below. I think both cases should give the same error, either the current error or the new error introduced in this PR. @MaryamZi WDYT?

`operator '&' not defined for 'future<(boolean|error)>' and 'future<(boolean|error)>'`

Reply:
`Producer1|Producer2` is a special case because it is similar to the alternate wait action (https://ballerina.io/spec/lang/master/#section_7.7.3), which is why that is special-cased.